src/menu/menu.c

```diff
@@ -314,7 +325,23 @@ handle_menu_element(xmlNode *n, struct server *server)
 
 	if (execute) {
 		wlr_log(WLR_ERROR, "we do not support pipemenus");
-	} else if (label && id) {
+	} else if ((label && id) || (id && nr_parents(n) == 2)) {
+		/*
+		 * (label && id) refers to <menu id="" label=""> which is an
+		 * inline menu definition.
+		 *
+		 * (id && nr_parents(n) == 2) refers to:
+		 * <openbox_menu>
+		 *   <menu id="">
+		 *   </menu>
+		 * </openbox>
+		 *
+		 * which is the highest level a menu can be defined at.
+		 * Whist openbox spec stricly speaking wants a label="" defined
```

Comment:
spelling

Otherwise: LGTM. Not tested though.

Reply:
Thanks :smile: